app/code/Magento/Inventory/Model/ResourceModel/Reservation/GetReservationsQuantity.php

```diff
@@ -45,7 +46,15 @@ public function execute(string $sku, int $stockId): float
             ->where(ReservationInterface::STOCK_ID . ' = ?', $stockId)
             ->limit(1);
 
-        $reservationQty = $connection->fetchOne($select);
+        try {
+            $reservationQty = $connection->fetchOne($select);
+        } catch (\Exception $e) {
```

Comment:
looks like this is the wrong place for catching the exception 
As we don't have exceptional situation at the time of Reservation fetching

Reply:
The issue happens in 

 Magento\InventoryIndexer\Model\GetStockItemData

Reply:
The exception was thrown in past if the index was not build for the stock